packages/shared/src/alternative.tms/alternative.tms.ts

```diff
@@ -0,0 +1,48 @@
+import { TileMatrixSet, TileMatrixSetType } from '@basemaps/geo';
+
+export class AlternativeTileMatrixSet extends TileMatrixSet {
+    altName: string;
+    parent: TileMatrixSet;
+    scaleMap: Map<number, number>;
+
+    constructor(def: TileMatrixSetType, parent: TileMatrixSet, altName: string) {
+        super(def);
+        this.parent = parent;
+        this.altName = altName;
+        this.scaleMap = new Map();
+
+        // Setup the z mapping between child Tmx and parent Tmx based on the scaleDenominator
+        let last = null;
+        for (const pTmx of this.parent.def.tileMatrix) {
+            for (const cTmx of this.def.tileMatrix) {
+                if (last == null) {
+                    if (cTmx.scaleDenominator >= pTmx.scaleDenominator) {
+                        this.scaleMap.set(Number(cTmx.identifier), Number(pTmx.identifier));
+                    }
+                } else if (this.parent.def.tileMatrix.indexOf(pTmx) == this.parent.def.tileMatrix.length - 1) {
```

Comment:
if you need to do `indexOf`, its best to switch this loop into  `for (let i = 0; i < ...) { ` style as `indexOf` is not a cheap operation

Reply:
Might be a idea to move this into a standalone function too, that returns a map, from two tile matrix sets